### PR TITLE
[Issue-746] Add unit tests for queue Commands

### LINKED ISSUE #746: Add unit tests for `queue` related commands
Queue commands:
  - `create queue`
  - `delete queue`
  - `describe queue`
  - `update queue`
  
  should be covered by unit tests.

### PR DESCRIPTION
**Pull request by steffnova**
[Problem]

Queue related commands are not covered with tests

[Solution]

Add queue package in pkg/client that abastracts GRPC calls to armada server.

All queue commands have function for their specific API call as injection.
Injected functions are from pkg/client/queue
parameter. This appraochs ensures that comamnd can be tested. Test cover
flag validation, argument validation and error handling. Command print
statments are not part of the tests.

Close #746

### CODE REVIEW

File: cmd/armadactl/cmd/queue/delete_test.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+			cmd.SetErr(io.Discard)
+
+			if err := cmd.Execute(); err == nil {
+				t.Errorf("failed to error")
+				return false
+			}
+
+			return true
+		},
+		"sucess": func(queueName string) bool {
```
Comment (robertdavidsmith) [inline cmd/armadactl/cmd/queue/delete_test.go:49]: sucess => success

File: cmd/armadactl/cmd/queue/describe_test.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+			cmd.SetErr(io.Discard)
+
+			if err := cmd.Execute(); err == nil {
+				t.Errorf("failed to error")
+				return false
+			}
+
+			return true
+		},
+		"sucess": func(queueName string, jobSetInfos []api.JobSetInfo) bool {
```
Comment (robertdavidsmith) [inline cmd/armadactl/cmd/queue/describe_test.go:51]: sucess=>success

**Review by robertdavidsmith — CHANGES_REQUESTED** [review]

File: cmd/armadactl/cmd/queue/delete_test.go (hunk 1 above)
Comment (severinson) [reply cmd/armadactl/cmd/queue/delete_test.go:49]:
```suggestion
		"success": func(queueName string) bool {
```

File: cmd/armadactl/cmd/queue/describe_test.go (hunk 2 above)
Comment (severinson) [reply cmd/armadactl/cmd/queue/describe_test.go:51]:
```suggestion
		"success": func(queueName string, jobSetInfos []api.JobSetInfo) bool {
```

**Review by severinson — CHANGES_REQUESTED** [review]
Correct spelling errors

**Review by robertdavidsmith — APPROVED** [review]

**Review by severinson — APPROVED** [review]

**Pull request merged**